terraform/main.tf

```
@@ -181,6 +181,33 @@ module "grants_prepared_data_bucket" {
   ]
 }
 
+module "ffis_source_data_bucket" {
+  source  = "cloudposse/s3-bucket/aws"
+  version = "3.0.0"
+  context = module.s3_label.context
+  name    = "ffis_source_data"
+
+  acl                          = "private"
+  versioning_enabled           = true
+  sse_algorithm                = "AES256"
+  allow_ssl_requests_only      = true
+  allow_encrypted_uploads_only = false
+  source_policy_documents      = [data.aws_iam_policy_document.ses_source_data_s3_access.json]
+
+  lifecycle_configuration_rules = [
+    {
+      enabled                                = true
+      id                                     = "rule-1"
+      filter_and                             = null
+      abort_incomplete_multipart_upload_days = 1
+      transition                             = [{ days = null }]
+      expiration                             = { days = 30 }
+      noncurrent_version_transition          = []
```

This was nagging at me from earlier so I checked and it does need to contain at least one object with `days = null`... otherwise terraform complains about ACLs (for some reason).
```suggestion
      noncurrent_version_transition          = [{ days = null }]
```

---

Noting for posterity...

This _may_ have not ultimately been necessary – turns out the ACL error is related to a [bug](https://github.com/cloudposse/terraform-aws-s3-bucket/issues/174) in the CloudPosse module, which currently has an [open PR](https://github.com/cloudposse/terraform-aws-s3-bucket/issues/174) that fixes the issue.

However, after re-running the initially [failed](https://github.com/usdigitalresponse/grants-ingest/actions/runs/4831799599/attempts/1) workflow, it did [succeed](https://github.com/usdigitalresponse/grants-ingest/actions/runs/4831799599), and the lifecycle configuration looks good in Staging 🎉

---

Huh, I was wondering why that failed :sweat_smile: